### PR TITLE
Add Helm functionality

### PR DESCRIPTION
**Pull request by mamercad**
Might help with #657; you can see how this works in my fork if you like:

```bash
❯ ./bin/helm repo add awx-operator https://mamercad.github.io/awx-operator/
"awx-operator" has been added to your repositories

❯ ./bin/helm repo update
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "awx-operator" chart repository
Update Complete. ⎈Happy Helming!⎈

❯ ./bin/helm search repo awx
NAME                            CHART VERSION   APP VERSION     DESCRIPTION                      
awx-operator/awx-operator       0.17.1          0.17.1          A Helm chart for the AWX Operator

❯ ./bin/helm install my-awx-operator awx-operator/awx-operator
NAME: my-awx-operator
LAST DEPLOYED: Thu Feb 17 22:09:05 2022
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Helm Chart 0.17.1

❯ ./bin/helm list
NAME            NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                   APP VERSION
my-awx-operator default         1               2022-02-17 22:09:05.825593 -0500 EST    deployed        awx-operator-0.17.1     0.17.1     

❯ kubectl -n awx get deploy
NAME                              READY   UP-TO-DATE   AVAILABLE   AGE
awx-operator-controller-manager   0/1     1            0           2m19s
```

The deployment isn't up-to-date because there's no `awx-operator:0.17.1` pushed up to quay right now.

### CODE REVIEW

**Comment by mamercad** [comment]
I forgot, this is how the usage looks:

```bash
❯ VERSION=0.17.1 make helm-chart
== KUSTOMIZE (image and namespace) ==
cd config/manager && /home/mark/src/github.com/mamercad/awx-operator/bin/kustomize edit set image controller=quay.io/ansible/awx-operator:0.17.1
cd config/default && /home/mark/src/github.com/mamercad/awx-operator/bin/kustomize edit set namespace awx
== HELM ==
cd charts && \
        /home/mark/src/github.com/mamercad/awx-operator/bin/helm create awx-operator --starter /home/mark/src/github.com/mamercad/awx-operator/.helm/starter ;\
        sed -i -e 's/version: 0.1.0/version: 0.17.1/' awx-operator/Chart.yaml ;\
        sed -i -e 's/appVersion: 0.1.0/appVersion: "0.17.1"/' awx-operator/Chart.yaml ;\
        sed -i -e 's/description: A Helm chart for Kubernetes/description: A Helm chart for the AWX Operator/' awx-operator/Chart.yaml
Creating awx-operator
apiVersion: v2
appVersion: "0.17.1"
description: A Helm chart for the AWX Operator
name: awx-operator
type: application
version: 0.17.1
== KUSTOMIZE (annotation) ==
cd config/manager && /home/mark/src/github.com/mamercad/awx-operator/bin/kustomize edit set annotation helm.sh/chart:awx-operator-0.17.1
cd config/default && /home/mark/src/github.com/mamercad/awx-operator/bin/kustomize edit set annotation helm.sh/chart:awx-operator-0.17.1
== SLICE ==
/home/mark/src/github.com/mamercad/awx-operator/bin/kustomize build config/default | \
        /home/mark/src/github.com/mamercad/awx-operator/bin/kubectl-slice --input-file=- \
                --output-dir=charts/awx-operator/templates \
                --sort-by-kind
Wrote charts/awx-operator/templates/namespace-awx.yaml -- 156 bytes.
Wrote charts/awx-operator/templates/serviceaccount-awx-operator-controller-manager.yaml -- 158 bytes.
Wrote charts/awx-operator/templates/configmap-awx-operator-awx-manager-config.yaml -- 472 bytes.
Wrote charts/awx-operator/templates/customresourcedefinition-awxbackups.awx.ansible.com.yaml -- 2685 bytes.
Wrote charts/awx-operator/templates/customresourcedefinition-awxrestores.awx.ansible.com.yaml -- 2740 bytes.
Wrote charts/awx-operator/templates/customresourcedefinition-awxs.awx.ansible.com.yaml -- 18122 bytes.
Wrote charts/awx-operator/templates/clusterrole-awx-operator-metrics-reader.yaml -- 216 bytes.
Wrote charts/awx-operator/templates/clusterrole-awx-operator-proxy-role.yaml -- 348 bytes.
Wrote charts/awx-operator/templates/clusterrolebinding-awx-operator-proxy-rolebinding.yaml -- 359 bytes.
Wrote charts/awx-operator/templates/role-awx-operator-awx-manager-role.yaml -- 1495 bytes.
Wrote charts/awx-operator/templates/role-awx-operator-leader-election-role.yaml -- 524 bytes.
Wrote charts/awx-operator/templates/rolebinding-awx-operator-awx-manager-rolebinding.yaml -- 374 bytes.
Wrote charts/awx-operator/templates/rolebinding-awx-operator-leader-election-rolebinding.yaml -- 382 bytes.
Wrote charts/awx-operator/templates/service-awx-operator-controller-manager-metrics-service.yaml -- 351 bytes.
Wrote charts/awx-operator/templates/deployment-awx-operator-controller-manager.yaml -- 1823 bytes.
```

File: .github/workflows/helm-release.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: helm-release
+
+on:
+  push:
+    branches:
+      - helm
```
Comment (mamercad) [inline .github/workflows/helm-release.yaml]: This'll have to be `devel` ...

File: Makefile [hunk 2]
```diff
@@ -103,7 +115,7 @@ ifeq (,$(shell which kustomize 2>/dev/null))
 	@{ \
 	set -e ;\
 	mkdir -p $(dir $(KUSTOMIZE)) ;\
-	curl -sSLo - https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize/v3.8.7/kustomize_v3.8.7_$(OS)_$(ARCH).tar.gz | \
+	curl -sSLo - https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize/v4.5.2/kustomize_v4.5.2_$(OS)_$(ARCH).tar.gz | \
```
Comment (mamercad) [inline Makefile]: Needed to be able to edit/set annotations.

**Review by JAORMX — APPROVED** [review]
We'd really use this chart! Thanks for kickstarting this

**Comment by mamercad** [comment]
> We'd really use this chart! Thanks for kickstarting this

No worries; Helm's my preferred way right now (so, I need it, heh). Also, you'll need to push a `gh-pages` branch and enable Pages functionality. It's a start anyhow, we can think about doing more Helm templating down the road.

**Comment by JAORMX** [comment]
Yeah, we use Helm all over, so I need this as well. Hopefully the community accepts this.

**Review by JAORMX — APPROVED** [review]

**Comment by mamercad** [comment]
Let me update the `README.md` with releasing information...

**Comment by mamercad** [comment]
I'll add "install using Helm" to the `README.md` next.

**Review by JAORMX — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -182,3 +194,62 @@ catalog-build: opm ## Build a catalog image.
...
+.PHONY: helm-chart
+helm-chart: kustomize helm kubectl-slice
+	@echo "== KUSTOMIZE (image and namespace) =="
+	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
+	cd config/default && $(KUSTOMIZE) edit set namespace ${NAMESPACE}
+
+	@echo "== HELM =="
+	cd charts && \
+		$(HELM) create awx-operator --starter $(shell pwd)/.helm/starter ;\
+		$(SED_I) -e 's/version: 0.1.0/version: $(VERSION)/' $(CHART_NAME)/Chart.yaml ;\
```
Comment (St0rmingBr4in) [inline Makefile]: Maybe using yq instead of sed would be preferable ? This adds a dependance on yq but ensure we have a valid yaml file at the end. Also we might want to fail if the command fails I think.

Comment (mamercad) [reply Makefile]: Pushed...

**Comment by shanemcd** [comment]
Hello. Thank you for taking the time to write and submit this PR. Right now I have some concerns that would need to be addressed before we would be able to consider merging this code.

We've put in a great deal of effort to remove any hardcoded versions or rendered files in the repo, enabling us to do releases at any time by running a simple workflow. We have a relatively small team and do not have the resources to constantly update files when we do a release.

As you can see in the Makefile, we pull the version from git:

https://github.com/ansible/awx-operator/blob/c02e05925e6dce17b0d9f768a516b86cc6a6734d/Makefile#L6

And all of the resources are generated dynamically by kustomize:

https://github.com/ansible/awx-operator/blob/c02e05925e6dce17b0d9f768a516b86cc6a6734d/Makefile#L89

Similarly to what I said in https://github.com/ansible/awx-operator/issues/657, maybe we could generate these files at release time.

**Comment by mamercad** [comment]
> Hello. Thank you for taking the time to write and submit this PR. Right now I have some concerns that would need to be addressed before we would be able to consider merging this code.
> 
> We've put in a great deal of effort to remove any hardcoded versions or rendered files in the repo, enabling us to do releases at any time by running a simple workflow. We have a relatively small team and do not have the resources to constantly update files when we do a release.
> 
> As you can see in the Makefile, we pull the version from git:
> 
> https://github.com/ansible/awx-operator/blob/c02e05925e6dce17b0d9f768a516b86cc6a6734d/Makefile#L6
> 
> And all of the resources are generated dynamically by kustomize:
> 
> https://github.com/ansible/awx-operator/blob/c02e05925e6dce17b0d9f768a516b86cc6a6734d/Makefile#L89
> 
> Similarly to what I said in #657, maybe we could generate these files at release time.

Sure, I can work on that.

**Comment by mamercad** [comment]
@shanemcd I think this is closer to what you're looking for, please have another look when you have some time.

**Comment by mamercad** [comment]
Using my fork; after pushing `0.17.3`, this [workflow](https://github.com/mamercad/awx-operator/actions/runs/1890484174) , followed by this [workflow](https://github.com/mamercad/awx-operator/actions/runs/1890485525) ran. The release is created and can be viewed [here](https://github.com/mamercad/awx-operator/releases/tag/awx-operator-0.17.3). Lastly, after updating my Helm repo, I can find the chart:

```bash
❯ helm search repo awx-operator
NAME                            CHART VERSION   APP VERSION     DESCRIPTION                      
awx-operator/awx-operator       0.17.3          0.17.3          A Helm chart for the AWX Operator
```

**Comment by pat-s** [comment]
Thanks for this!

I've tried to install it but saw the following

```
Error: INSTALLATION FAILED: rendered manifests contain a resource that already exists. Unable to continue with install: Namespace "awx" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "awx"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "default"
```

Web search indicates that this error is quite generic and hence I am not sure if it really relates to the helm chart. Wanted to post it anyway...

**Comment by mamercad** [comment]
> Thanks for this!
> 
> I've tried to install it but saw the following
> 
> ```
> Error: INSTALLATION FAILED: rendered manifests contain a resource that already exists. Unable to continue with install: Namespace "awx" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "awx"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "default"
> ```
> 
> Web search indicates that this error is quite generic and hence I am not sure if it really relates to the helm chart. Wanted to post it anyway...

Seems like you might have either AWX or the AWX Operator already deployed? I just tried this again, here's [a gist](https://gist.github.com/mamercad/1e781e661728ca411f61362bea1e0bb6).

**Comment by shanemcd** [comment]
This is looking pretty great @mamercad. Thank you for all your work here. Apologies for being slow to respond, we've been been busy lately. I want to test this out, but I'm on vacation most of next week so it might be the week of the 14th before I can give this a proper review.

**Comment by mamercad** [comment]
> This is looking pretty great @mamercad. Thank you for all your work here. Apologies for being slow to respond, we've been been busy lately. I want to test this out, but I'm on vacation most of next week so it might be the week of the 14th before I can give this a proper review.

No worries at all; thanks for taking a look -- I'm in no rush on this whatsoever, I can always use my fork if needed -- have a nice vacation!

File: .github/workflows/helm-release.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+name: helm-release
+
+on:
+  push:
+    tags: ['*']
```
Comment (shanemcd) [inline .github/workflows/helm-release.yaml]: Can you change to match this?:

https://github.com/ansible/awx-operator/blob/6b873b05abe37f39d90a8594abc54a899d210b06/.github/workflows/promote.yaml#L4-L5

File: Makefile [hunk 5]
```diff
@@ -182,3 +198,123 @@ catalog-build: opm ## Build a catalog image.
...
+	rmdir bin/$(OS)-$(ARCH) ;\
+	}
+else
+HELM = $(shell which helm)
+endif
+endif
+
+.PHONY: yq
+YQ = $(shell pwd)/bin/yq
+helm: ## Download helm locally if necessary.
```
Comment (shanemcd) [inline Makefile]: Is this supposed to be `yq:`? I'm seeing this: 

```
Makefile:241: warning: overriding recipe for target 'helm'
Makefile:223: warning: ignoring old recipe for target 'helm'
```

File: Makefile [hunk 6]
```diff
@@ -94,6 +109,7 @@ undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/confi
 
 OS := $(shell uname -s | tr '[:upper:]' '[:lower:]')
 ARCH := $(shell uname -m | sed -e 's/x86_64/amd64/' -e 's/aarch64/arm64/')
+ARCH2 := $(shell uname -m | sed -e 's/aarch64/arm64/')
```
Comment (shanemcd) [inline Makefile]: Why is this needed? This looks like it should output the same thing as the line above it. What am I missing?

**Review by shanemcd — COMMENTED** [review]
Amazing work on this. I left some minor comments just after cloning and reading the code. I haven't yet tried it out on my fork. Will do that sometime next week. Thanks again, this is a great contribution.

File: Makefile (hunk 6 above)
Comment (mamercad) [reply Makefile]: My mind must have been broken for a minute :)

File: Makefile (hunk 5 above)
Comment (mamercad) [reply Makefile]: Nice catch!

**Comment by mamercad** [comment]
> Amazing work on this. I left some minor comments just after cloning and reading the code. I haven't yet tried it out on my fork. Will do that sometime next week. Thanks again, this is a great contribution.

No worries; appreciate the review!

**Comment by shanemcd** [comment]
@mamercad I just sat down to finally test this and noticed I just merged a PR that created a merge conflict. Would you mind resolving that? Or I can handle it and push to your branch. Let me know what you prefer.

**Comment by mamercad** [comment]
> @mamercad I just sat down to finally test this and noticed I just merged a PR that created a merge conflict. Would you mind resolving that? Or I can handle it and push to your branch. Let me know what you prefer.

I can rebase, give me a bit.

**Comment by shanemcd** [comment]
I'm also seeing this:

```
$ make kubectl-slice

gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
make: *** [Makefile:205: kubectl-slice] Error 2
```

It looks like this is because on my machine `ARCH` is `amd64` which does not map to a filename under https://github.com/patrickdappollonio/kubectl-slice/releases/tag/v1.1.0

**Comment by mamercad** [comment]
@shanemcd I think this is ready for another 👀

**Comment by shanemcd** [comment]
Oh the sweet game of whack-a-mole 😄:

```
$ make yq
tar: yq_linux_amd64: Not found in archive
tar: Exiting with failure status due to previous errors
make: *** [Makefile:240: yq] Error 2
```

**Comment by shanemcd** [comment]
I'm wondering if it might be prudent to add a helm-chart CI check. Ideally it would do a deploy if possible, but ensure it can build at very least.

**Comment by mamercad** [comment]
> I'm wondering if it might be prudent to add a helm-chart CI check. Ideally it would do a deploy if possible, but ensure it can build at very least.

Yep, that's a good idea; I'll work on it.

**Comment by mamercad** [comment]
@shanemcd Curious what you think about the proposed `helm-test` workflow when you have some time.

**Comment by shanemcd** [comment]
Looks like the original CI checks are failing with `Error: unknown flag: --load_restrictor"`

**Comment by mamercad** [comment]
> Looks like the original CI checks are failing with `Error: unknown flag: --load_restrictor"`

Hrm; I can't seem to find that output, got a link or snippet you could share?

**Comment by no-12** [comment]
> > Looks like the original CI checks are failing with `Error: unknown flag: --load_restrictor"`
> 
> Hrm; I can't seem to find that output, got a link or snippet you could share?

https://github.com/ansible/awx-operator/runs/5874981454?check_suite_focus=true#step:6:4084

Line 4084 of the "run Molecule" job output

**Comment by mamercad** [comment]
> Looks like the original CI checks are failing with `Error: unknown flag: --load_restrictor"`

I think it's because underscores changed to hyphens in `kustomize` [v4.0.1](https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv4.0.1) to be compatible with `kubectl`; I've pushed the change.

**Comment by shanemcd** [comment]
CI checks are passing, nice work! Give me a few more days, I still need to run through a test release on my fork. This should get merged by sometime early next week based on my current priorities.

**Comment by mamercad** [comment]
> CI checks are passing, nice work! Give me a few more days, I still need to run through a test release on my fork. This should get merged by sometime early next week based on my current priorities.

Yep, sounds good, as always, thanks for looking!

**Comment by shanemcd** [comment]
Hi @mamercad - sincere apologies for the delay here, I got derailed chasing a bug in another project for a while.

I tried running a release on my fork and saw this:

```
/home/runner/work/awx-operator/awx-operator/bin/cr upload \
	--owner "ansible" \
	--git-repo "awx-operator" \
	--token "***" \
	--skip-existing
Error: error creating GitHub release awx-operator-0.20.1: POST https://api.github.com/repos/ansible/awx-operator/releases: 403 Resource not accessible by integration []
```

I think we need to override this from the workflow:

https://github.com/ansible/awx-operator/blob/cf6b04297544551d7c95467d01af44428f8454ff/Makefile#L54

See `${{ github.repository_owner }}`, which we use in a few other places:

https://github.com/ansible/awx-operator/blob/7fd5083c16142950f92f0949b25b3c3b615353d0/.github/workflows/stage.yml#L65

**Comment by mamercad** [comment]
> Hi @mamercad - sincere apologies for the delay here, I got derailed chasing a bug in another project for a while.
> 
> I tried running a release on my fork and saw this:
> 
> ```
> /home/runner/work/awx-operator/awx-operator/bin/cr upload \
> 	--owner "ansible" \
> 	--git-repo "awx-operator" \
> 	--token "***" \
> 	--skip-existing
> Error: error creating GitHub release awx-operator-0.20.1: POST https://api.github.com/repos/ansible/awx-operator/releases: 403 Resource not accessible by integration []
> ```
> 
> I think we need to override this from the workflow:
> 
> https://github.com/ansible/awx-operator/blob/cf6b04297544551d7c95467d01af44428f8454ff/Makefile#L54
> 
> See `${{ github.repository_owner }}`, which we use in a few other places:
> 
> https://github.com/ansible/awx-operator/blob/7fd5083c16142950f92f0949b25b3c3b615353d0/.github/workflows/stage.yml#L65

No problem at all; changes pushed!

**Comment by shanemcd** [comment]
Now I am seeing this. Any ideas?

```
2022-04-27T13:06:02.1003033Z == CHART RELEASER (index) ==
2022-04-27T13:06:02.1005421Z /home/runner/work/awx-operator/awx-operator/bin/cr index \
2022-04-27T13:06:02.1005951Z 	--owner "shanemcd" \
2022-04-27T13:06:02.1007505Z 	--git-repo "awx-operator" \
2022-04-27T13:06:02.1008339Z 	--token "***" \
2022-04-27T13:06:02.1008788Z 	--pages-branch "gh-pages" \
2022-04-27T13:06:02.1009330Z 	--index-path "./charts/index.yaml" \
2022-04-27T13:06:02.1011777Z 	--charts-repo "https://shanemcd.github.io/awx-operator/index.yaml" \
2022-04-27T13:06:02.1012500Z 	--remote httpsorigin \
2022-04-27T13:06:02.1012938Z 	--push
2022-04-27T13:06:02.2587128Z UpdateIndexFile new index at ./charts/index.yaml
2022-04-27T13:06:02.4084397Z Found awx-operator-20.1.0.tgz
2022-04-27T13:06:02.4085440Z Extracting chart metadata from .cr-release-packages/awx-operator-20.1.0.tgz
2022-04-27T13:06:02.4091185Z Calculating Hash for .cr-release-packages/awx-operator-20.1.0.tgz
2022-04-27T13:06:02.4092408Z Updating index ./charts/index.yaml
2022-04-27T13:06:02.4113252Z fatal: invalid reference: httpsorigin/gh-pages
```

**Comment by mamercad** [comment]
> Now I am seeing this. Any ideas?
> 
> ```
> 2022-04-27T13:06:02.1003033Z == CHART RELEASER (index) ==
> 2022-04-27T13:06:02.1005421Z /home/runner/work/awx-operator/awx-operator/bin/cr index \
> 2022-04-27T13:06:02.1005951Z 	--owner "shanemcd" \
> 2022-04-27T13:06:02.1007505Z 	--git-repo "awx-operator" \
> 2022-04-27T13:06:02.1008339Z 	--token "***" \
> 2022-04-27T13:06:02.1008788Z 	--pages-branch "gh-pages" \
> 2022-04-27T13:06:02.1009330Z 	--index-path "./charts/index.yaml" \
> 2022-04-27T13:06:02.1011777Z 	--charts-repo "https://shanemcd.github.io/awx-operator/index.yaml" \
> 2022-04-27T13:06:02.1012500Z 	--remote httpsorigin \
> 2022-04-27T13:06:02.1012938Z 	--push
> 2022-04-27T13:06:02.2587128Z UpdateIndexFile new index at ./charts/index.yaml
> 2022-04-27T13:06:02.4084397Z Found awx-operator-20.1.0.tgz
> 2022-04-27T13:06:02.4085440Z Extracting chart metadata from .cr-release-packages/awx-operator-20.1.0.tgz
> 2022-04-27T13:06:02.4091185Z Calculating Hash for .cr-release-packages/awx-operator-20.1.0.tgz
> 2022-04-27T13:06:02.4092408Z Updating index ./charts/index.yaml
> 2022-04-27T13:06:02.4113252Z fatal: invalid reference: httpsorigin/gh-pages
> ```

Can you push a branch named `gh-pages`?

**Comment by shanemcd** [comment]
Hmmm. I _can_, but what should that branch contain? Are there any docs I can read about the approach you're taking here? I probably should have asked that question sooner 🙂

**Comment by mamercad** [comment]
> Hmmm. I _can_, but what should that branch contain? Are there any docs I can read about the approach you're taking here? I probably should have asked that question sooner 🙂

Yep, it's using [chart releaser](https://github.com/helm/chart-releaser), which operates on a [branch](https://github.com/ansible/awx-operator/pull/799/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R315). It's been a while since I've looked at this as well at this point, but, I think it's better to use a separate branch for GitHub Pages (versus the default branch of the project), and that by default, is a branch named `gh-pages`. I don't think that `cr` can create the branch if it doesn't exist.

**Comment by shanemcd** [comment]
Looking into it now, not sure I'll figure it out before I get pulled off to something else.

After reading a bit it looks like it's expecting a clean slate with a "charts" directory. I'm going to try that first.

**Comment by shanemcd** [comment]
Hope you dont mind me pushing to this branch. Found a small little problem in your Makefile changes. We probably didn't see it in GHA since their runners already have yq installed.

**Comment by mamercad** [comment]
> Hope you dont mind me pushing to this branch. Found a small little problem in your Makefile changes. We probably didn't see it in GHA since their runners already have yq installed.

Not at all :) -- we'll (re)figure out how this works, together, hehe.

**Comment by nickjmv** [comment]
We are really looking forward to an official helm chart to use. Can I conclude that you guys are close to a helm chart installation method for the operator?

**Comment by Hokwang** [comment]
@mamercad
I am the person who is looking forward to using helm chart of awx-operator and moreover I think it is ok to remove kustomize. helm chart can be replacement.

By the way, I am looking in to code, the values.yaml is empty. no one can adjust values. I think this is just wrapper,
user can run only `helm install awx-operator` without `-f dev-values.yaml` or `-f prod-values.yaml`.

and I think whole helm chart code should exist in codebase. In your code, chart tgz file is generated only. not easy to see diff.

**Comment by shanemcd** [comment]
>  I think it is ok to remove kustomize. helm chart can be replacement.

We can't do this because operator-sdk (and surrounding tooling like OLM) is built on top of an auto-generated kustomize-based framework. And from what I can tell it seems to be more flexible than helm.

> By the way, I am looking in to code, the values.yaml is empty. no one can adjust values

This is a pretty fair complaint. I dont know much about helm, but do these values require you to use helm's templating language?

**Comment by shanemcd** [comment]
@mamercad I rebased and resolved a conflict with this PR + pushed 2 new commits that:

- adds a basic smoke test in CI
- gets this integrated into our existing release processes

The comment from @Hokwang is valid and makes me question the overall usefulness of this, but I guess it's better than nothing and will work for the general case. We can continue to iterate on it if there are good solutions to these problems.

**Comment by mamercad** [comment]
> @mamercad I rebased and resolved a conflict with this PR + pushed 2 new commits that:
> 
> * adds a basic smoke test in CI
> * gets this integrated into our existing release processes
> 
> The comment from @Hokwang is valid and makes me question the overall usefulness of this, but I guess it's better than nothing and will work for the general case. We can continue to iterate on it if there are good solutions to these problems.

Totally agree; it's just a start and a framework. If folks find it useful going forward, we can put work towards the Helm values.

**Comment by nickjmv** [comment]
So right now there is no option to modify values for the helm chart? You just have to take what the chart deploys, right? Or is there a workaround for now when the upstream values.yaml file is still empty?

**Comment by mamercad** [comment]
> So right now there is no option to modify values for the helm chart? You just have to take what the chart deploys, right? Or is there a workaround for now when the upstream values.yaml file is still empty?

This was *just* merged; so no, there's no workaround right now. As it stands right now, it packages up [the existing deployment method](https://github.com/ansible/awx-operator/blob/bf74d5cc342c34f6c8ddba2171828b20cd36c42b/Makefile#L291), i.e., `kustomize build`.

**Comment by mateuszdrab** [comment]
@mamercad Thanks for contributing this in

Quick question, could you provide some steps to safely migrate from the kustomize deployment to helm without losing the existing instances? Would be nice if this could be added to the upgrade section of the project's readme.
I'm fearing that if I delete the kustomize deployment and associated CRDs, I will lose my instance

**Comment by mamercad** [comment]
> @mamercad Thanks for contributing this in
> 
> Quick question, could you provide some steps to safely migrate from the kustomize deployment to helm without losing the existing instances? Would be nice if this could be added to the upgrade section of the project's readme. I'm fearing that if I delete the kustomize deployment and associated CRDs, I will lose my instance

Unfortunately, I can't give advice on this other than "caveat emptor". If it were me, I'd **definitely** test this in a sandbox before production. If I can get some time, I'll try this in my homelab (but, to be honest, I've been swamped with work lately).

**Comment by mateuszdrab** [comment]
No rush, I'll try to deploy it when I gain more 'confidence' - I think I'll just remove all resources except the crds and redeploy via helm (which should take ownership of the CRDs then)

**Comment by mamercad** [comment]
> No rush, I'll try to deploy it when I gain more 'confidence' - I think I'll just remove all resources except the crds and redeploy via helm (which should take ownership of the CRDs then)

Yeah, that sounds reasonable. Not sure how closely you [looked](https://github.com/ansible/awx-operator/blob/3ac0232e89e3fc5a1aa5358ad84d54bb3957d982/Makefile#L291) -- the Helm chart is simply packaging up the result of `kustomize build` (after slicing up by kind).
